.travis.yml

```diff
@@ -2,10 +2,21 @@ language: php
 
 php:
   - 7.0
+  - 7.1
+
+env:
+  matrix:
+    - PREFER_LOWEST="--prefer-lowest"
+    - PREFER_LOWEST=""
+
+cache:
+  directories:
+    - $HOME/.composer/cache
 
 before_script:
-  - composer self-update
-  - composer install --no-interaction
+  - travis_retry composer self-update
+  - travis_retry composer update --no-interaction --prefer-source $PREFER_LOWEST

```

Comment:
Why not `install` rather than update ? Also, I would use a `--prefer-dist` instead of sources (as we won't edit), and add `--no-suggest`, `--optimize-autoloader` and `--classmap-authoritative`.

Reply:
I’m using --prefer-source as a dist version will ignore the tests/ directory (as it is excluded from export in .gitattributes file). It prevents also tests from randomly failing if you run out of limited amount of GitHub API requests.